Docs: Add guidance for custom User type in Auth guides (#7354)
edgedb/edgedb: docs/guides/auth/email_password.rst

   };

> Creating a User object
> ----------------------

Reviewer: ```suggestion
Create a User object
--------------------
```
Make verb tense consistent with other section headings

> Some applications may want to create a custom ``User`` type in their default
> module to attach application-specific information. You can tie this to an
> ``ext::auth::Identity`` by using the ``auth_token`` in our
> ``ext::auth::client_token`` global and inserting your ``User`` object with a
> link to the ``Identity``.

Reviewer: ```suggestion
For some applications, you may want to create a custom ``User`` type in the
default module to attach application-specific information. You can tie this to
an ``ext::auth::Identity`` by using the ``auth_token`` in our
``ext::auth::client_token`` global and inserting your ``User`` object with a
link to the ``Identity``.
```
I know this is a nitpick, but it feels off to me to say that an application "wants" something. Tweaked the language just a bit to instead talk about what the developer wants.

> For this example, we'll assume you have a 1-to-1 relationship between ``User``
> objects and ``ext::auth::Identity`` objects, but in your own application, you
> may decide to have a 1-to-many relationship.

Reviewer: ```suggestion
.. note::

    For this example, we'll assume you have a one-to-one relationship between
    ``User`` objects and ``ext::auth::Identity`` objects. In your own
    application, you may instead decide to have a one-to-many relationship.
```

This feels less like a core piece of the content and more like a note about the upcoming example. Made some other minor tweaks too.

> So, given this ``User`` type:

Reviewer: ```suggestion
Given this ``User`` type:
```
